cylc/uiserver/resolvers.py

```diff
@@ -316,9 +316,18 @@ async def play(cls, workflows, args, workflows_mgr, log):
 
                 if ret:
                     # command failed
-                    _, err = proc.communicate()
+                    out, err = proc.communicate()
+                    if 'previously run with' in out:
+                        # Can't upgrade:
+                        msg = out
+                    else:
+                        msg = (
+                            f'Could not start {tokens["workflow"]}'
+                            f' - {cmd_repr}'
+                        )
+
                     raise Exception(
-                        f'Could not start {tokens["workflow"]} - {cmd_repr}'
+                        msg
                         # suppress traceback unless in debug mode
                         + (f' - {err}' if DEBUG else '')
                     )
```

Comment:
IMHO this whole thing could be
```
msg = out.strip() or err.strip() or (
    f'Could not start {tokens["workflow"]}'
    f' - {cmd_repr}'
)
raise Exception(
    msg
    # include possible traceback in debug mode
    + (f' - {err}' if DEBUG and not out.strip() else '')
)
```
Better to include stderr if no stdout is available IMHO

Reply:
The whole point of this code is to do something different if the stdout matches a given pattern? I'll ask in person.